components/camel-ai/camel-langchain4j-agent-api/src/main/java/org/apache/camel/component/langchain4j/agent/api/guardrails/JsonFormatGuardrail.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.langchain4j.agent.api.guardrails;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.guardrail.OutputGuardrail;
+import dev.langchain4j.guardrail.OutputGuardrailResult;
+
+/**
+ * Output guardrail that validates AI responses are valid JSON format.
+ *
+ * <p>
+ * This guardrail is useful when you expect structured JSON responses from the LLM. It can validate basic JSON syntax,
+ * required fields, and optionally extract the JSON portion from a response that contains additional text.
+ * </p>
+ *
+ * <p>
+ * Example usage:
+ * </p>
+ *
+ * <pre>{@code
+ * JsonFormatGuardrail guardrail = JsonFormatGuardrail.builder()
+ *         .requireFields("name", "email", "age")
+ *         .extractJson(true)
+ *         .build();
+ * }</pre>
+ *
+ * @since 4.17.0
+ */
+public class JsonFormatGuardrail implements OutputGuardrail {
```

Comment:
This is really cool!

Reply:
Yes, this come to my mind because in some response I had missing field in the json..